rm/rm-server/src/test/java/functionaltests/nodesource/TestNodeSourceThreadPool.java

```diff
@@ -65,27 +73,24 @@
     @Before
     public void setup() throws Exception {
         TestSSHInfrastructureV2.startSSHServer();
-        String rmconf = new File(RMTHelper.class.getResource(CONFIG_FILE_PATH).toURI()).getAbsolutePath();
-        this.rmHelper.startRM(rmconf);
-        this.resourceManager = this.rmHelper.getResourceManager();
     }
 
     @Test
     public void testDownNodesCanStillBeDetectedWhenDeploymentThreadsAreExhausted() throws Exception {
-
+        String rmConfig = new File(RMTHelper.class.getResource(TWO_THREADS_CONFIG_FILE_PATH).toURI()).getAbsolutePath();
+        this.rmHelper.startRM(rmConfig);
         this.resourceManager = this.rmHelper.getResourceManager();
-
         this.resourceManager.defineNodeSource(NODE_SOURCE_NAME,
                                               SSHInfrastructureV2.class.getName(),
                                               getInfiniteRetryInfrastructureParameters(),
                                               StaticPolicy.class.getName(),
-                                              TestSSHInfrastructureV2.policyParameters,
+                                              getStaticPolicyParameters(),
                                               NODES_NOT_RECOVERABLE);
         this.resourceManager.deployNodeSource(NODE_SOURCE_NAME);
 
         RMTHelper.log("Waiting for the RM to have one free node");
         while (this.resourceManager.getState().getFreeNodesNumber() != ONE_NODE_PER_HOST) {
-            Thread.sleep(500);
+            Thread.sleep(ACTIVE_WAITING_PERIOD);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)